bin/oil.py

```diff
@@ -192,15 +192,17 @@ def OshMain(argv0, argv, login_shell):
       exec_opts.interactive = True
   elif opts.i:  # force interactive
     arena.PushSource('<stdin -i>')
-    line_reader = reader.InteractiveLineReader(OSH_PS1, arena)
+    prompt = ui.Prompt(OSH_PS1, arena, parse_ctx, ex)
+    line_reader = reader.InteractiveLineReader(arena, prompt)
     exec_opts.interactive = True
   else:
     try:
       script_name = arg_r.Peek()
     except IndexError:
       if sys.stdin.isatty():
         arena.PushSource('<interactive>')
-        line_reader = reader.InteractiveLineReader(OSH_PS1, arena)
+        prompt = ui.Prompt(OSH_PS1, arena, parse_ctx, ex)
```

Comment:
The OSH_PS1 constant can be just moved into ui.py as `DEFAULT_PS1` or something.  It doesn't really need to be a member since it's a constant.